lib/dangermattic/plugins/ios_release_check.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module Danger
+  # Plugin for performing iOS / macOS release-related checks in a pull request.
+  #
+  # @example Checking for changes in Core Data models on a release branch:
+  #          ios_release_check.check_core_data_model_changed
+  #
+  # @example Checking for modified Localizable.strings on a regular branch:
+  #          ios_release_check.check_modified_localizable_strings
+  #
+  # @example Checking for synchronization between release notes and App Store strings:
+  #          ios_release_check.check_release_notes_and_app_store_strings
+  #
+  # @see Automattic/dangermattic
+  # @tags ios, macos, process, release
+  #
+  class IosReleaseCheck < Plugin
+    LOCALIZABLE_STRINGS_FILE = 'Localizable.strings'
+    BASE_STRINGS_FILE = "en.lproj/#{LOCALIZABLE_STRINGS_FILE}".freeze
+
+    # Checks if an existing Core Data model has been edited in a release branch.
+    #
+    # @return [void]
+    def check_core_data_model_changed
+      warning = 'Do not edit an existing Core Data model in a release branch unless it hasn\'t been released to testers yet. ' \
+                'Instead create a new model version and merge back to develop soon.'
+
+      common_release_checks.check_file_changed(
+        file_comparison: ->(path) { File.extname(path) == '.xcdatamodeld' },
+        message: warning,
+        on_release: true
+      )
+    end
+
+    # Checks if any Localizable.strings file has been modified on a release branch, otherwise reporting a warning.
+    #
+    # @return [void]
+    def check_modified_localizable_strings_on_release
+      common_release_checks.check_file_changed(
+        file_comparison: ->(path) { File.basename(path) == LOCALIZABLE_STRINGS_FILE },
+        message: "The `#{LOCALIZABLE_STRINGS_FILE}` files should only be updated on release branches, when the translations are downloaded.",
+        on_release: false
+      )
+    end
+
+    # Checks if the en.lproj/Localizable.strings file has been modified on a regular branch, otherwise reporting a warning.
+    #
+    # @return [void]
+    def check_modified_en_strings_on_regular_branch
+      common_release_checks.check_file_changed(
+        file_comparison: ->(path) { path.end_with?(BASE_STRINGS_FILE) },
+        message: "The `#{BASE_STRINGS_FILE}` file should only be updated before creating a release branch.",
+        on_release: true
+      )
+    end
+
+    # Checks if a translation file (*.lproj/Localizable.strings) has been modified on a release branch, otherwise reporting a warning.
+    #
+    # @return [void]
+    def check_modified_translations_on_release_branch
+      common_release_checks.check_file_changed(
+        file_comparison: ->(path) { !path.end_with?(BASE_STRINGS_FILE) && File.basename(path) == LOCALIZABLE_STRINGS_FILE },
```

Comment:
Same feedback as before wrt the use of `end_with?` to compare file paths.

Prefer using `File.basename(path) != BASE_STRINGS_FILE` instead of `!path.end_with?(BASE_STRINGS_FILE)` to avoid matching unexpected files like `fi-en.lproj/Localizable.strings` or whatnot (not a valid locale code here, but you see my point)

Reply:
Though `File.basename(path) != BASE_STRINGS_FILE` will always be true, even when `path = en.lproj/Localizable.strings`, since `File.basename(path)` will return `Localizable.strings`, right?
Perhaps just using `path != BASE_STRINGS_FILE` is best, but I'm not sure if `path` can have other elements that would break the check (like `./` at the start).

Haven't tested it, but what if we do
```ruby
BASE_STRINGS_FILE_PATTERN = "*/en.lproj/#{LOCALIZABLE_STRINGS_FILE}"
[...]
{ !File.fnmatch(BASE_STRINGS_FILE_PATTERN, path) && File.basename(path) == LOCALIZABLE_STRINGS_FILE }
```

Reply:
Maybe a helper method that would check if `path == LOCALIZABLE_STRINGS_FILE || path.end_with?('/' + LOCALIZABLE_STRINGS_FILE.delete_prefix('/'))`, to account for both the case of if `LOCALIZABLE_STRINGS_FILE` happen to be rooted at the root of the repo (hence not being in any subdirectory, and thus not having a leading `/` if it's already the top dir) and adding a `/` prefix if not present already for other cases…

Though I think implementing such logic like that could still make us miss some edge cases (like paths containing `..` and stuff that we should maybe run `File.realpath` or `Pathname.cleanpath` on them first. Actually, I'm wondering if there isn't any method from `Pathname` (which is a bit feature-rich compared to `File.…` methods when manipulating paths) that would already be able to do that for us?

Reply:
Digging the docs, maybe something like this would work:

```
base_path_name = Pathname.new(BASE_STRINGS_FILE)
Pathname.new(path).last(base_path_name.length) == base_path_name
```

Reply:
Wfm. To be extracted in a helper method to give it a nice name.

Reply:
Updated it: 0b8318e6ecc51e32618c600db95707367d24c2ec